llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_nvidia.py

```diff
@@ -0,0 +1,221 @@
+
+        chat_response = llm.chat([message])
+        assert chat_response.message.content == "Cool Test Message"
+
+
+@patch("llama_index.llms.nvidia.base.SyncOpenAI")
+def test_chat_model_streaming(MockSyncOpenAI: MagicMock) -> None:
+    with CachedNVIDIApiKeys(set_fake_key=True):
+        mock_instance = MockSyncOpenAI.return_value
+        mock_instance.chat.completions.create.return_value = (
+            mock_chat_completion_stream_v1()
+        )
+
+        llm = NVIDIA()
+        prompt = "test prompt"
+        message = ChatMessage(role="user", content="test message")
+
+        response_gen = llm.stream_complete(prompt)
+        responses = list(response_gen)
+        assert responses[-1].text == "TestSecond Test"
+
+        mock_instance.chat.completions.create.return_value = (
+            mock_chat_completion_stream_v1()
+        )
+
+        chat_response_gen = llm.stream_chat([message])
+        chat_responses = list(chat_response_gen)
+        print(chat_responses)
+        assert chat_responses[-1].message.content == "TestSecond Test"
+        assert chat_responses[-1].message.role == "assistant"
+
+
+@pytest.mark.asyncio()
+@patch("llama_index.llms.nvidia.base.AsyncOpenAI")
+async def test_async_chat_model_basic(MockAsyncOpenAI: MagicMock) -> None:
+    with CachedNVIDIApiKeys(set_fake_key=True):
+        mock_instance = MockAsyncOpenAI.return_value
+        create_fn = AsyncMock()
+        create_fn.side_effect = mock_async_chat_completion_v1
+        mock_instance.chat.completions.create = create_fn
+
+        llm = NVIDIA()
+        prompt = "test prompt"
+        message = ChatMessage(role="user", content="test message")
+
+        response = await llm.acomplete(prompt)
+        assert response.text == "Cool Test Message"
+
+        chat_response = await llm.achat([message])
+        assert chat_response.message.content == "Cool Test Message"
+
+
+@pytest.mark.asyncio()
+@patch("llama_index.llms.nvidia.base.AsyncOpenAI")
+async def test_async_streaming_chat_model(MockAsyncOpenAI: MagicMock) -> None:
+    with CachedNVIDIApiKeys(set_fake_key=True):
+        mock_instance = MockAsyncOpenAI.return_value
+        create_fn = AsyncMock()
+        create_fn.side_effect = mock_async_chat_completion_stream_v1
+        mock_instance.chat.completions.create = create_fn
+
+        llm = NVIDIA()
+        prompt = "test prompt"
+        message = ChatMessage(role="user", content="test message")
+
+        response_gen = await llm.astream_complete(prompt)
+        responses = [response async for response in response_gen]
+        assert responses[-1].text == "TestSecond Test"
+
+        chat_response_gen = await llm.astream_chat([message])
+        chat_responses = [response async for response in chat_response_gen]
+        assert chat_responses[-1].message.content == "TestSecond Test"
+
+
+def test_validates_api_key_is_present() -> None:
+    with CachedNVIDIApiKeys(set_fake_key=True):
```

Comment:
`NVIDIA()` should work with no api_key or NVIDIA_API_KEY env to support users doing `NVIDIA().mode("nim", base_url=...)` without a key. please add a test for that case.